cpp/pybind/core/hashmap.cpp

```diff
@@ -105,8 +106,33 @@ void pybind_core_hashmap(py::module& m) {
 
     hashmap.def("to", &Hashmap::To, "device"_a, "copy"_a = false);
     hashmap.def("clone", &Hashmap::Clone);
-    hashmap.def("cpu", &Hashmap::CPU);
-    hashmap.def("cuda", &Hashmap::CUDA, "device_id"_a = 0);
+    hashmap.def(
+            "cpu",
+            [](const Hashmap& hashmap) {
+                return hashmap.To(
+                        core::Device(core::Device::DeviceType::CPU, 0));
+            },
+            "Transfer the hashmap to CPU. If the hashmap "
+            "is already on CPU, no copy will be performed.");
+    hashmap.def(
+            "cuda",
+            [](const Hashmap& hashmap, int device_id) {
+                if (!core::cuda::IsAvailable()) {
+                    utility::LogError(
+                            "CUDA is not available, cannot copy Hashmap.");
+                }
+                if (device_id < 0 || device_id >= core::cuda::DeviceCount()) {
+                    utility::LogError(
+                            "Invalid device_id {}, must satisfy 0 <= "
+                            "device_id < {}",
+                            device_id, core::cuda::DeviceCount());
+                }
+                return hashmap.To(core::Device(core::Device::DeviceType::CUDA,
```

Comment:
`core::Device("CUDA", device_id)`, the first argument can be a string. 
Apply this change to all `DeviceType::CUDA` and `DeviceType::CPU` in this PR,